### PR TITLE
Some updates to fix Errors

### PR DESCRIPTION
**Pull request by Fehdra**
1.1.3.0 to 1.2.4.0 in the main

changes to prevent a NullReferenceException i was having so added  && vrm != null) in ValheimVRM.cs

Fixed some game errors for VRMEyePositionSync

### CODE REVIEW

File: ValheimVRM/ValheimVRM.cs [hunk 1]
```diff
@@ -376,31 +376,41 @@ static void Postfix(Humanoid __instance, Ragdoll ragdoll)
...
-				else
-				{
-					lodGroup.localReferencePoint = new Vector3(999999f, 999999f, 999999f);
-				}
-			}
-		}
-	}
-
-	[HarmonyPatch(typeof(Player), "OnDeath")]
+	// To prevent a NullReferenceException, you should check if vrm is not null before attempting to access its components. Here's the modified code with a null check: var lodGroup = vrm.GetComponent<LODGroup>();
```
Comment (aMidnightNova) [inline ValheimVRM/ValheimVRM.cs]: Move this notes block to the PR notes.

File: ValheimVRM/ValheimVRM.cs [hunk 2]
```diff
@@ -376,31 +376,41 @@ static void Postfix(Humanoid __instance, Ragdoll ragdoll)
...
+
+    [HarmonyPatch(typeof(Character), "SetVisible")]
+    static class Patch_Character_SetVisible
+    {
+        [HarmonyPostfix]
+        static void Postfix(Character __instance, bool visible)
+        {
+            if (!__instance.IsPlayer()) return;
+
+            if (VrmManager.PlayerToVrmInstance.TryGetValue((Player)__instance, out var vrm) && vrm != null)
```
Comment (aMidnightNova) [inline ValheimVRM/ValheimVRM.cs]: you can put the here a note about "null check before accessing components"

File: ValheimVRM/ValheimVRM.cs [hunk 3]
```diff
@@ -376,31 +376,41 @@ static void Postfix(Humanoid __instance, Ragdoll ragdoll)
...
+            {
+                var lodGroup = vrm.GetComponent<LODGroup>();
+                if (lodGroup != null)
```
Comment (aMidnightNova) [inline ValheimVRM/ValheimVRM.cs]: Same for here about the new null check for log group.

E.G. // avoid reference errors use null check for lod.

**Review by aMidnightNova — CHANGES_REQUESTED** [review]
Overall looks good, just a few minor things. I will test this later today or tomorrow.

**Review by Fehdra — COMMENTED** [review]
Like this?  :}

**Comment by aMidnightNova** [comment]
Everything seams to work, merging.

**Pull request merged**